coordinator/src/main/scala/filodb.coordinator/NodeClusterActor.scala

```diff
@@ -314,14 +311,10 @@ private[filodb] class NodeClusterActor(settings: FilodbSettings,
     case GetShardMap(ref)       => shardManager.sendSnapshot(ref, sender())
   }
 
-  // The initial recovery handler: recover dataset setup/ingestion config first
   def datasetHandler: Receive = LoggingReceive {
     case e: SetupDataset =>
-      setupDataset(e, sender()) map { _ => self ! SetupDatasetFinished(e.ref) }
-    case e: SetupDatasetFinished => {
-      initDatasets -= e.ref
-      if (initDatasets.isEmpty) initiateShardStateRecovery()
-    }
+      // used by unit testing only
+      initializeDataset(e.dataset, e.config, Some(sender()))
```

Comment:
Why do we even need the `SetupDataset` anymore?  Can we get rid of it entirely?  Tests should be using config too.  We shouldn't have a test-only path here.  Please make the tests use the same init mechanism as everyone else....

Reply:
Fixing unit tests to use config was a big change and introduced possibility of regressions. Hence retained the SetupDataset capability there. 

Multi-JVM test like IngestionAndRecoverySpec use the same path as the real servers. They do not use SetupDataset.